helm/release-operator/templates/configmap.yaml

```diff
@@ -14,6 +14,8 @@ data:
       listen:
         address: 'http://0.0.0.0:8000'
     service:
+      app:
+        extraAnnotations: '{{ .Values.Installation.V1.GiantSwarm.AppOperator.PauseAnnotation }}:true'
```

Comment:
This is also confusing to me - what is the goal of this PR?

Won't this stop reconciliation of all apps created by release operator?

Reply:
We can add `config-controller.giantswarm.io/version:0.0.0` here as well. That way they won't.

Reply:
@MarcelMue @stone-z The aim of this issue and PR is to stop App CR reconciliation until `config-controller` generates configmap/secret for them and remove the annotation.

@kopiczko Not with current setup - it'll just kick it into a loop. Maybe adding a case in config-controller where it removes the `paused` annotation in case the App CR is missing `version` annotation.

Reply:
If this is the goal, then it seems like a bad idea to me to do it this way.

This here will cause the annotation to be added back in _every_ reconciliation loop of `release-operator`. Release operator also handles EVERY release in EVERY loop because of the union / cut calculations we have to do. IMO this would cause every app to become paused for the majority of time.

Why does the config controller not add the pause annotation and remove it then? Is a race between app-operator and config-controller the issue?

Reply:
Thinking more about it: It would probably not do it on every loop with the current implementation, because we do not compare objects closely currently.

It still introduces a hidden workflow - adding a non spec field as an annotation which from this operators PoV makes no sense. I can already see someone in the future removing this annotation here because it's not obvious that you should deploy new apps as `paused`.

Reply:
> Why does the config controller not add the pause annotation and remove it then? Is a race between app-operator and config-controller the issue?

Yes.

> Thinking more about it: It would probably not do it on every loop with the current implementation, because we do not compare objects closely currently.
> 
> It still introduces a hidden workflow - adding a non spec field as an annotation which from this operators PoV makes no sense. I can already see someone in the future removing this annotation here because it's not obvious that you should deploy new apps as `paused`.

That's a good point.

We need to rethink the approach.